Content.Shared/Movement/Components/MovementIgnoreGravityComponent.cs

```diff
@@ -36,6 +38,13 @@ public static bool IsWeightless(this EntityUid entity, PhysicsComponent? body =
 
             mapManager ??= IoCManager.Resolve<IMapManager>();
             var grid = mapManager.GetGrid(gridId);
+            var invSys = EntitySystem.Get<InventorySystem>();
+
+            if (invSys.TryGetSlotEntity(entity, "shoes", out var ent))
+            {
+                if (entityManager.TryGetComponent<SharedMagbootsComponent>(ent, out var boots) && boots.On)
+                    return false;
+            }
```

Comment:
This check is already expensive af (it's 1/3 of physics time for reference) and really shouldn't be here.

Realistically it should apply the component that's above for this  (though you also need to check it doesn't give them space movement).

Reply:
Screw it I'll deal with it later when I refactor this anyway.